### DISCUSSION TITLE
How I enable to get trained policy of Isaac-Repose-Cube-Allegro-v0 task?

### DISCUSSION THREAD

**Discussion by cold-young** [isaac-sim/IsaacLab#1442] [Q&A]
Hi there, 

I have been trying to train the Isaac-Repose-Cube-Allegro-v0 task using the skrl PPO algorithm. 
(num_envs 2048, <1 hour)
Even though the reinforcement learning reward graph has converged, I cannot get an expert policy. 
I think I need well defined shaped rewards or other learning approaches like imitation learning.
By the way, I found a good demonstration of in-hand manipulation with shadow hand. [on this page](https://github.com/isaac-sim/IsaacLab/discussions/1008)
To acquire expert policy, what i do ? 

Thanks

**Comment by RandomOakForest** [comment]
Thanks for posting this. By expert policy, do you mean the final trained policy? You may find them in this directory:
```
./IsaacLab/logs/<learning_library_name>/<task_name>
```

**Reply by cold-young** [reply]
Thank you for your answer.

I mean, I cannot get a well-trained policy using the "manager_based" Isaac-Repose-Cube-Allegro-v0 task.
![image](https://github.com/user-attachments/assets/c8b58252-d59b-46e6-9bfe-9e00c694f6e1)

I was able to acquire expert policy using "Isaac-Repose-Cube-Allegro-Direct-v0" task. 
![image](https://github.com/user-attachments/assets/605af6d9-7e0f-4ab8-97db-83bb3e552d75)

Although the "manager_based" task is useful to adjust many parameters (in reward and states), I cannot get expert policy. (Maybe I should change many parameters in the Isaac-Repose-Cube-Allegro-v0 task). 
How do I do for getting well-trained policy using "Isaac-Repose-Cube-Allegro-v0" task.
Default setting is not work to train

**Comment by StrainFlow** [comment]
I think the first step is for us to reproduce what you're seeing. Can you give me reproduction steps for your two policies so I can see if I get the same results? Are you using the samples found here directly: 

https://isaac-sim.github.io/IsaacLab/main/source/overview/environments.html

Are these also the shadow hand examples you're looking for?

**Reply by cold-young** [reply]
Hi, 

I used exactly this example without any changes.
"./isaaclab.sh -p source/standalone/workflows/skrl/train.py --task Isaac-Repose-Cube-Allegro-v0 --num_envs 4096 --headless" and 
"./isaaclab.sh -p source/standalone/workflows/skrl/train.py --task Isaac-Repose-Cube-Allegro-Direct-v0 --num_envs 4096 --headless"

I think "Isaac-Repose-Cube-Allegro-Direct-v0" has well-defined parameters (rewards), but "Isaac-Repose-Cube-Allegro-v0" doesn't. 
And I found the radomisation of "Isaac-Repose-Cube-Allegro-v0" is more varied including mass, texture. 

Thanks for your help.

**Comment by jeffyu880** [comment]
Hi @cold-young Were you able to determine how to get the Allegro-v0 task to give a good policy? I have run into the same issue, where the policy does not perform well with the default training parameters. I increased the timesteps and the mean reward is not any better. As seen in the image below

<img width="511" height="448" alt="image" src="https://github.com/user-attachments/assets/482287bc-c72d-46ca-a6de-42a6402b7483" />
